lib/auth/jwtheader.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var jws = require('jws');
+
+/**
+ * JWTHeader service account credentials.
+ *
+ * Create a new access token by using the credential to create a new JWT token
+ * that's recognized as the access token.
+ *
+ * @param {string=} email the service account email address.
+ * @param {string=} key the private key that will be used to sign the token.
+ * @constructor
+ */
+function JWTHeader(email, key) {
+  this.email = email;
+  this.key = key;
+}
+
+/**
+ * Executes the given callback if it is not null.
+ *
+ * Is used to simplify invocation of optional callbacks.
+ */
+function maybeCall_(c, err, res) {
```

Comment:
This seems like a general purpose helper. Perhaps make it internal or move to a utility class.